server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -122,14 +127,44 @@ public class RemoteClusterStateService implements Closeable {
         Metadata::fromXContent
     );
 
+    public static final ChecksumBlobStoreFormat<CoordinationMetadata> COORDINATION_METADATA_FORMAT = new ChecksumBlobStoreFormat<>(
```

Comment:
This class has to be refactored it has become too huge and union of remote metadata handling across different types, we will need to work on it immediately after this PR.

Reply:
Agreed Shweta, I have a draft for this refactor ready which I will raise immediately after this merge https://github.com/shiv0408/OpenSearch/pull/7